packages/expo-modules-core/android/src/main/java/expo/modules/interfaces/taskManager/TaskServiceProviderHelper.kt

```diff
@@ -0,0 +1,36 @@
+package expo.modules.interfaces.taskManager
+
+import android.content.Context
+import expo.modules.core.ModulePriorities
+import expo.modules.core.interfaces.Package
+
+class TaskServiceProviderHelper {
+  companion object {
+    /*
+      Uses reflection to look through the current list of packages and attempts to find
+      one that provides a TaskServiceInterface implementation.
+     */
+    fun getTaskServiceImpl(context: Context): TaskServiceInterface? {
+      // Use reflection to get the packages list from ExpoModulesPackageList without
+      // creating the reactInstanceManager:
+      val expoModules: Class<*>? = try {
+        Class.forName("expo.modules.ExpoModulesPackageList")
```

Comment:
I have a concern about the current implementation. The package list may not reflect what users are actually using and might not even exist in the project. For example, in brownfield applications, users might not utilize our autolinking. Therefore, if the package list is missing, we should attempt to create our task manager directly.

Reply:
Agree - But that's where the dependency issue comes in and gives us trouble. `expo-modules-core` (or the consuming package like `expo-notifications`) would have to depend on `expo-task-manager` and then we break the loose coupling. 

Maybe `expo-notifications` should emit a warning about the problem and suggest that we install `expo-task-manager` ? Or even add `expo-task-manager` as a NPM dependency in `expo-notifications`?

Reply:
I’m sorry for the wording in my previous comment; it wasn’t ideal. The word “directly” doesn’t fit. I meant to build it through reflection; just use the task manager implementation class rather than the package list.

Reply:
See your point! I thought that I'd try to not introduce things that might break so I found this already existing usage of the package list - it is a build-time generated class that is only accessed in this way - and if it is changed a lot of other stuff will break so it will be very visible. Adding a second lookup through reflection on the task service by its class string might not picked up as clearly. In addition if we change the implementation of the `TaskService` but keep the interface this will still work?